test/PostgreSQL/Test.hs

```diff
@@ -1227,6 +1231,192 @@ testCommonTableExpressions = do
             pure res
         asserting $ vals `shouldBe` fmap Value [2..11]
 
+testPostgresqlLocking :: SpecDb
+testPostgresqlLocking =
+    -- These tests use a create a fork in order to run two transactions concurrently.
```

Comment:
I could possibly abstract this forking behavior out into a function but doing so feels a bit messy given the need for different queries and assertions. Please let me know if that would be strongly preferred though!

Reply:
Nah, I think this is fine - any generalization would have to take into account all kinds of weird behavior, so it's not clear the best way to abstract the behavior.